client/platform/desktop/components/Recent.vue

```diff
@@ -93,23 +95,31 @@ export default defineComponent({
             </h2>
             <div
               v-for="recent in recents"
-              :key="recent.dir + recent.base"
+              :key="recent.id"
               class="pa-1"
             >
               <h3 class="text-body-1">
                 <v-icon
                   class="pr-2"
                   color="primary lighten-2"
                 >
-                  {{ recent.ext ? 'mdi-file-video' : 'mdi-folder-open' }}
+                  {{
+                    (recent.type === 'video')
+                      ? 'mdi-file-video'
+                      : (recent.originalImageFiles.length > 1)
```

Comment:
This will fail on the initial migration.  You'll have a list of recents but the items won't have the `originalImageFiles` property so when attempting to find the length it will crash and lock up on first launch from the old version to the new one.  May need to do a guard against that.  I know you're looking for an excuse to get rid of it and I don't really want to give you one.

Reply:
I don't understand.

> Also the first migration crashing because of recent.originalImageFiles.length

I think this might just be a race.  The migration should erase localStorage if it isn't properly formatted, but it's async and needs to block the vue app starting.